internal/pkg/manifest/svc.go

```diff
@@ -495,6 +495,46 @@ func (rr RoutingRuleConfiguration) exposedPorts(exposedPorts []ExposedPort, work
 	}
 }
 
+// exportPorts returns any new ports that should be exposed given the network load balancer
+// configuration that's not part of the existing containerPorts.
+func (cfg NetworkLoadBalancerConfiguration) exposedPorts(exposedPorts []ExposedPort, workloadName string) ([]ExposedPort, error) {
+	if cfg.IsEmpty() {
+		return nil, nil
+	}
+	containerName := workloadName
+	if cfg.TargetContainer != nil {
+		containerName = aws.StringValue(cfg.TargetContainer)
+	}
+	nlbPort, protocol, err := ParsePortMapping(cfg.Port)
+	if err != nil {
+		return nil, err
+	}
+	if protocol == nil {
+		protocol = aws.String("tcp")
+	}
+
+	port, err := strconv.ParseUint(aws.StringValue(nlbPort), 10, 16)
+	if err != nil {
+		return nil, err
+	}
+	containerPort := uint16(port)
+	if cfg.TargetPort != nil {
+		containerPort = uint16(aws.IntValue(cfg.TargetPort))
+	}
+	for _, exposedPort := range exposedPorts {
+		if containerPort == exposedPort.Port && cfg.TargetContainer == nil {
+			return nil, nil
+		}
+	}
+	return []ExposedPort{
+		{
+			Port:          containerPort,
```

Comment:
qq: why do we keep `Port` to be `uint16` instead of `string`?

Reply:
There is no solid reason behind keeping it as `uint16` but we accept primary container's port as `uint16` so I thought I should keep it consistent and use as much as `uint16` everywhere so that there won't frequent conversion issue going forward from `uint16` to string and vice a versa.